addons/purchase_stock/models/purchase.py

```diff
@@ -304,6 +304,18 @@ def write(self, values):
             self.filtered(lambda l: l.order_id.state == 'purchase')._create_or_update_picking()
         return result
 
+    def unlink(self):
+        for line in self:
+            line.move_ids._action_cancel()
```

Comment:
@DonatasV You're right for the move_dest_ids I was focus on move_ids and forget move_dest_ids.
in order to gain performance I suggest you to make all those steps in batch (if you cancel multiple line in one transaction). It could be something like this:
```python
def unlink(self):
    line.move_ids._action_cancel()

    ppg_cancel_lines = self.filtered(lambda line: line.propagate_cancel)
    ppg_cancel_lines.move_dest_ids._action_cancel()

    not_ppg_cancel_lines = self.filtered(lambda line: not line.propagate_cancel)
    not_ppg_cancel_lines.move_dest_ids.write({'procure_method': 'make_to_stock'})
    not_ppg_cancel_lines.move_dest_ids.recompute_state()

Reply:
@Whenrow Thanks. I apply your suggestion

Reply:
@Whenrow @DonatasV I think the call to function `recompute_state()` is wrong, since the function is called `_recompute_state()`
Here a call is made to the function with the correct name (https://github.com/odoo/odoo/blob/14.0/addons/purchase_stock/models/purchase.py#L117).
The bug has been detected in Odoo 14.0 runbot

![odoo_runbot_14](https://user-images.githubusercontent.com/29223264/117985779-48c04500-b339-11eb-9e17-6c081f115f03.png)

Reply:
The wrong method is called is actually in [13.0 branch](https://github.com/odoo/odoo/blame/13.0/addons/purchase_stock/models/purchase.py#L315) , and is causing big troubles
As ioans says it should be called _recompute_state as [is defined](https://github.com/odoo/odoo/blob/13.0/addons/stock/models/stock_move.py#L1507)

Reply:
This typo is already fixed here https://github.com/odoo/odoo/pull/70800. Sorry for the inconvenience